lib/pavilion/series.py

```diff
@@ -155,6 +161,38 @@ def _save_series_id(self):
             self._logger.warning("Could not save series id to '%s'",
                                  last_series_fn)
 
+    def _sys_name_series_tracking(self):
+        """Save the series id to json file that tracks last series ran by user
+        on a per system basis."""
+
+        sys_vars = system_variables.get_vars(True)
+        sys_name = sys_vars['sys_name']
+
+        json_file = self.pav_cfg.working_dir/'users'
+        json_file /= '{}.json'.format(utils.get_login())
+
+        lockfile_path = json_file.parent/(json_file.name + '.lock')
+        lockfile = LockFile(lockfile_path)
```

Comment:
I didn't lock in the original function because it wasn't necessary. In this case though it is only necessary to prevent loss of data, not file conflicts. Because you're writing a new file every time, you never have to worry about corrupting the read of some other process. You may have a situation like this though:

- proc A reads
- proc B reads
- proc A writes
- proc B writes

In which case, the final file will be missing the data changed by proc A. 

You may already know all this, but I thought it would be good to point out.